### PR TITLE
TEST-#6893: Added support for 'pytest 8.0.0'

### LINKED ISSUE #6893: CI fails after release of pytest 8.0.0
https://github.com/modin-project/modin/actions/runs/7697508576/job/20974753158?pr=6862

the following code fails starting with pytest 8.0.0 ([example in our tests](https://github.com/modin-project/modin/blob/fe19363d58dfd50834971ee56e5038a822374594/modin/pandas/test/test_series.py#L217-L224))
```python
import pytest
import warnings

try:
    with pytest.warns(UserWarning, match="Warning"):
        raise TypeError("Hello")
        warnings.warn("Warning", UserWarning)
        x = 10
except TypeError:
    pass

# Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
#  Emitted warnings: [].
```

### PR DESCRIPTION
**Pull request by dchigarev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #6893 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/pandas/test/test_io_exp.py [hunk 1]
```diff
@@ -126,8 +126,8 @@ def test_read_csv_empty_frame(self):
         df_equals(modin_df, pandas_df)
 
     def test_read_csv_without_glob(self):
-        with pytest.warns(UserWarning, match=r"Shell-style wildcard"):
-            with pytest.raises(FileNotFoundError):
+        with pytest.raises(FileNotFoundError):
+            with warns_that_defaulting_to_pandas():
```
Comment (dchigarev) [inline modin/experimental/pandas/test/test_io_exp.py:130]: it actually raises default-to-pandas warning

File: modin/pandas/test/test_general.py [hunk 2]
```diff
@@ -295,15 +295,15 @@ def test_merge_asof_bad_arguments():
     modin_left, modin_right = pd.DataFrame(left), pd.DataFrame(right)
 
     # Can't mix by with left_by/right_by
-    with pytest.raises(ValueError), warns_that_defaulting_to_pandas():
+    with pytest.raises(ValueError):
```
Comment (dchigarev) [inline modin/pandas/test/test_general.py:298]: this was trying to catch default-to-pandas warnings of a pure pandas code (impossible)

File: modin/experimental/pandas/test/test_io_exp.py (hunk 1 above)
Comment (anmyachev) [reply modin/experimental/pandas/test/test_io_exp.py:130]: How did this work before?

Comment (dchigarev) [reply modin/experimental/pandas/test/test_io_exp.py:130]: I'm not sure, it seems that it just captured every `UserWarnings` here (including default-to-pandas warning) and proceed with no errors. In the new version of pytest, the default-to-pandas warning has to be captured individually for some reason

**Review by anmyachev — APPROVED** [review]
LGTM!

**Comment by anmyachev** [comment]
@dchigarev one more question. Should we pin `pytest>=8.0.0`?

**Pull request merged**